components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOImpl.java

```diff
@@ -0,0 +1,320 @@
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                int webhookId = addWebhookToDB(webhook, tenantId);
+                addWebhookEventsToDB(webhookId, webhook.getEventsSubscribed());
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw new WebhookMgtServerException("Error while creating the webhook in the system.", e);
+        }
+    }
+
+    @Override
+    public void updateWebhook(Webhook webhook, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                updateWebhookInDB(webhook, tenantId);
+                int webhookId = getInternalWebhookIdByUuid(webhook.getUuid(), tenantId);
+                deleteWebhookEventsInDB(webhookId);
+                addWebhookEventsToDB(webhookId, webhook.getEventsSubscribed());
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw new WebhookMgtServerException("Error while updating the webhook in the system.", e);
+        }
+    }
+
+    @Override
+    public void deleteWebhook(String webhookId, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            jdbcTemplate.withTransaction(template -> {
+                template.executeUpdate(WebhookSQLConstants.Query.DELETE_WEBHOOK,
+                        statement -> {
+                            statement.setString(WebhookSQLConstants.Column.UUID, webhookId);
+                            statement.setInt(WebhookSQLConstants.Column.TENANT_ID, tenantId);
+                        });
+                return null;
+            });
+        } catch (TransactionException e) {
+            throw new WebhookMgtServerException("Error while deleting the webhook in the system.", e);
+        }
+    }
+
+    @Override
+    public Webhook getWebhook(String webhookId, int tenantId) throws WebhookMgtException {
+
+        NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        try {
+            return jdbcTemplate.withTransaction(template -> {
+                Webhook webhook = template.fetchSingleRecord(WebhookSQLConstants.Query.GET_WEBHOOK_BY_ID,
+                        (resultSet, rowNumber) -> mapResultSetToWebhook(resultSet),
+                        statement -> {
+                            statement.setString(WebhookSQLConstants.Column.UUID, webhookId);
+                            statement.setInt(WebhookSQLConstants.Column.TENANT_ID, tenantId);
+                        });
+
+                if (webhook == null) {
+                    return null;
+                }
+
+                int internalWebhookId = getInternalWebhookIdByUuid(webhookId, tenantId);
```

Comment:
here also we should get the webhook using uuid and the tenant id

Reply:
https://github.com/wso2/carbon-identity-framework/pull/6741#discussion_r2100710901